src/App.tsx

```diff
@@ -1,39 +1,94 @@
 import React from 'react';
 import './App.scss';
 
+import Box from '@mui/material/Box';
+import Paper from '@mui/material/Paper';
+import LinearProgress from '@mui/material/LinearProgress';
+
+import '@fontsource/roboto/400.css';
+import '@fontsource/roboto/700.css';
+
+import { Clock } from './components/Clock/Clock';
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+export class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  timerId = 0;
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    document.addEventListener('click', this.handleClick);
+    document.addEventListener('contextmenu', this.handleContextmenu);
+  }
 
-        {' time is '}
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+    document.removeEventListener('click', this.handleClick);
+    document.removeEventListener('contextmenu', this.handleContextmenu);
+  }
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
-      </div>
-    </div>
-  );
-};
+  handleClick = () => {
+    this.setState({ hasClock: true });
+  };
+
+  handleContextmenu = (event: MouseEvent) => {
+    event.preventDefault();
+
+    this.setState({ hasClock: false });
+  };
+
+  render() {
+    const { clockName, hasClock } = this.state;
+
+    return (
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'center',
+          p: '200px',
+        }}
+      >
+        <Paper
+          elevation={3}
+          sx={{
+            backgroundColor: '#aaa',
+            px: 20,
+            py: 10,
+            transition: 'all 0.5s',
+            '&:hover': {
+              backgroundColor: '#eaa',
+            },
+          }}
+        >
+          <div className="App">
+            <h1>React clock</h1>
+
+            {hasClock && <Clock clockName={clockName} />}
+            {!hasClock && (
```

Comment:
Don't forget about breaks between elements
```suggestion
            {hasClock && <Clock clockName={clockName} />}
            
            {!hasClock && (
```